src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,37 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface TodoModalProps {
+  todo: Todo;
+  handleSetSelectedTodoId: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  handleSetSelectedTodoId,
+  todo,
+}) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(todo.userId)
+      .then(userData => {
+        setUser(userData);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [todo.userId]);
```

Comment:
You are directly interacting with the DOM by using the `getUser` function inside the `useEffect` hook. In React, you should avoid direct DOM manipulation as much as possible and let React handle the updates to the DOM. Consider moving the `getUser` function call to a separate function and call that function inside the `useEffect` hook.